src/backend/distributed/planner/multi_logical_planner.c

```diff
@@ -1387,62 +1388,52 @@ ExtractFromExpressionWalker(Node *node, QualifierWalkerContext *walkerContext)
 bool
 IsJoinClause(Node *clause)
 {
-	bool isJoinClause = false;
 	OpExpr *operatorExpression = NULL;
-	List *argumentList = NIL;
-	Node *leftArgument = NULL;
-	Node *rightArgument = NULL;
-	Node *strippedLeftArgument = NULL;
-	Node *strippedRightArgument = NULL;
+	bool equalsOperator = false;
+	List *varList = NIL;
+	Var *initialVar = NULL;
+	Var *var = NULL;
 
 	if (!IsA(clause, OpExpr))
 	{
 		return false;
 	}
 
-	operatorExpression = (OpExpr *) clause;
-	argumentList = operatorExpression->args;
+	operatorExpression = castNode(OpExpr, clause);
+	equalsOperator = OperatorImplementsEquality(operatorExpression->opno);
 
-	/* join clauses must have two arguments */
-	if (list_length(argumentList) != 2)
+	if (!equalsOperator)
 	{
+		/* Not an equi-join */
 		return false;
```

Comment:
Maybe add a comment on why we still need this restriction.

It would be nice if we could lift the it at some point, because it does not seem relevant whether or not the expression contains an "=" symbol. For instance `distance(dist.point, ref.point) < 20` is not fundamentally different from `mod((dist.x * dist.y),10000) = ref.z` in terms of the conditions in which it can be planned and executed.

Reply:
Note that removing doesn't seem to lead to any issues, a few queries in regression tests start working.

Reply:
I like the idea of supporting more, however we need to add this check back in Local and Partition joins as they still would rely on equality of both sides.

Biggest problem is testing the support of such joins, as it opens up a whole class of new queries that might work, or not :).

Reply:
removing the clause is tracked in https://github.com/citusdata/citus/issues/3198

Reply:
For now, a possible workaround is `(<expression>)::int = 1`